### ISSUE TITLE
A/C Pressure sensor considerations and use

### ISSUE THREAD

**Issue by ElDominio** [rusefi/rusefi#6570]
**Is your feature request related to a problem? Please describe.**
Most cars have A/C pressure sensors that let us know if the A/C pressure is too high (to shut off the compressor to not get into an overpressure situation), or too low (meaning there is a gas leak and we should not enable the compressor). We should be able to interpret voltage values from this 0-5v sensor and use it to control compressor state.

**Would we need new configuration fields?**
 calibration values for A/C pressure transducer
A/C Pressure Low Point Voltage
A/C Pressure High Point Voltage
A/C Low Point value - in kpa
A/C High point value - in kpa
Pressure Low disable - value of A/C pressure in kpa that compressor is disengaged
Pressure high disable - ""

**Describe the solution you'd like**
Add these calibration points to disable the compressor if bounds are exceeded

**Comment by ElDominio** [comment]
Also cousin of #4054

**Comment by kifir23917** [comment]
@ElDominio, what are the minimum and maximum values in kPa that we want to use for `Pressure Low disable` and `Pressure high disable` settings?

**Comment by ElDominio** [comment]
Bounds should be 0 and 500 kpa

Defaults should be 100 and 300 minimum maximum respectively

**Comment by kifir23917** [comment]
The functionality is ready for manual testing:
![Screnshot - AC Pressure Sensor](https://github.com/rusefi/rusefi/assets/159958110/e4b2b2cc-7d48-4f03-945d-400b768aae51)
![Screnshot - Air Conditioning](https://github.com/rusefi/rusefi/assets/159958110/b1b3f359-2063-4c2e-9181-83e14a161d2f)
Automated tests are still in work.

**Comment by kifir23917** [comment]
@ElDominio, A/C controller uses **deadbands** to prevent rapid switching on/off of AC. To check if A/C pressure is out of limits I use **deadband** with width = 5 (as we so for `Max CLT(deg C)`, `Max TPS(%)`). It means that if `Pressure Low disable(kPa)` is `200 kPa`, than A/C will be turned off when pressure is less than `200 - 5 = 195 kPa` and turned on back only after pressure exceeds `200 + 5 = 205 kPa`. Is it OK or do we need to reduce deadband width for `Pressure Low disable(kPa)` and `Pressure High disable(kPa)` settings?

> Bounds should be 0 and 500 kpa

It's obvious that `Pressure Low disable(kPa)` setting will never turn off A/C at if its value does not exceed deadband width (`5` in current implementation). So I'm going to use `5 kPa` as minimal acceptable value for `Pressure Low disable(kPa)` setting. Does it make sense for you?

**Comment by Zazins** [comment]
Hi!
Could you please add option with PWM Fan cotroll. As for now only can be controlled by PWM outputs and cant be set to on by AC. Its commonly used in cars to use PWM fans.
Thanks!

**Comment by rusefillc** [comment]
@Zazins I would have to remove your comment simply because it does not belong on "A/C Pressure sensor consideration" ticket

@Zazins feel free to create a separate ticket

**Comment by ElDominio** [comment]
ac pressure has no gauge, so no way to corroborate

**Comment by kifir23917** [comment]
I've added gauges for A/C Pressure:
![Screenshot - AC Pressure Sensor](https://github.com/rusefi/rusefi/assets/159958110/0e47e43d-7579-48be-ae92-667d8000cda1)

**Comment by ElDominio** [comment]
Confirmed working with nice gauges! Can close

**Comment by ElDominio** [comment]
Hysteresis needs to be configruable! I have noticed a hardcoded hysteresis of 5kpa, i need to be able to decide if i want to wait X amount of KPA until reenable

**Comment by rusefillc** [comment]
@ElDominio let me know if you expect me to reopen this ticket (I would rather not since it was technically a success) or if you need help creating new follow-up ticket?
